src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1591,6 +1592,31 @@ public void validateDeleteRequest(Dataset dataset, List<DataDeletionTableModel>
         }
     }
 
+    /*
+     * WARNING: Ensure SQL is validated before executing this method!
+     */
+    public List<Map<String, Object>> getSnapshotTableData(Snapshot snapshot,
+                                                          String sql) throws InterruptedException {
+        // execute query and get result
+        final BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        final TableResult result = bigQueryProject.query(sql);
+
+        // aggregate into single object
+        final FieldList columns = result.getSchema().getFields();
+        final List<Map<String, Object>> values = new ArrayList<>();
+        result.iterateAll().forEach(rows -> {
+            final Map<String, Object> rowData = new HashMap<>();
+            columns.forEach(column -> {
+                String columnName = column.getName();
+                Object fieldValue = rows.get(columnName).getValue();
+                rowData.put(columnName, fieldValue);
+            });
```

Comment:
I find the streaming version of this slightly easier to read. Also I think `rowData` is a great place to use `var` since we don't need to declare the type again.
```suggestion
            final var rowData =
                columns.stream()
                    .map(Field::getName)
                    .collect(Collectors.toMap(Function.identity(),
                        columnName -> rows.get(columnName).getValue()));
```

Reply:
I switched to using this approach.